homeassistant/components/omnilogic/omnilogic_common.py

```diff
@@ -0,0 +1,119 @@
+"""Common classes and elements for Omnilogic Integration."""
+
+from datetime import timedelta
+import logging
+
+from omnilogic import OmniLogicException
+
+from homeassistant.const import ATTR_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import ATTR_IDENTIFIERS, ATTR_MANUFACTURER, ATTR_MODEL, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class OmniLogicUpdateCoordinator(DataUpdateCoordinator):
+    """Class to manage fetching update data from single endpoint."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        api: str,
+        name: str,
+        polling_interval: int,
+    ):
+        """Initialize the global Omnilogic data updater."""
+        self.api = api
+
+        super().__init__(
+            hass=hass,
+            logger=_LOGGER,
+            name=name,
+            update_interval=timedelta(seconds=polling_interval),
+        )
+
+    async def _async_update_data(self):
+        """Fetch data from OmniLogic."""
+        try:
+            data = await self.api.get_telemetry_data()
+
+        except OmniLogicException as error:
+            raise UpdateFailed("Error updating from OmniLogic: %s" % error) from error
+
+        return data
+
+
+class OmniLogicEntity(CoordinatorEntity):
+    """Defines the base OmniLogic entity."""
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        kind: str,
+        name: str,
+        backyard: dict,
+        bow: dict,
+        icon: str,
+        entity_data: dict,
+    ):
+        """Initialize the OmniLogic Entity."""
+        super().__init__(coordinator)
+
+        backyard_name = backyard["BackyardName"].replace(" ", "_")
+        if bow != {}:
+            bow_name = bow["Name"].replace(" ", "_")
+
+            entity_name = f"{backyard_name}_{bow_name}_{kind}"
+        else:
+            entity_name = f"{backyard_name}_{kind}"
+
+        self._kind = kind
+        self._name = name
+        self._unique_id = entity_name
+        self._backyard = backyard
+        self._backyard_name = backyard["BackyardName"]
+        self._state = None
```

Comment:
We're not using `self._state` since we're calculating the state in the `state` property. We can remove this instance attribute and just use a local variable in the property.

Reply:
Addressed